src/cloud-api-adaptor/podvm/files/etc/systemd/system/kata-agent.service

```diff
@@ -1,15 +1,15 @@
 [Unit]
 Description=Kata Agent
 BindsTo=netns@podns.service
-Wants=process-user-data.service
-After=netns@podns.service process-user-data.service
+Wants=process-user-data-update.service
+After=netns@podns.service process-user-data-update.service
```

Comment:
regarding the idea discussed in https://github.com/confidential-containers/cloud-api-adaptor/issues/1852#issuecomment-2152135964, could we instead change the logic to provision the `agent-config.file` via cloud-config entry, so we don't have to perform any "updates"?